flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/AbstractQueryableStateTestBase.java

```diff
@@ -1342,7 +1347,7 @@ public void close() throws Exception {
                             Time.milliseconds(50),
                             deadline,
                             (jobStatus) -> jobStatus.equals(JobStatus.CANCELED),
-                            TestingUtils.defaultScheduledExecutor());
+                            new ScheduledExecutorServiceAdapter(EXECUTOR_RESOURCE.getExecutor()));
```

Comment:
We have so many locations where we put a `ScheduledExecutorServiceAdapter`. Couldn't we move this somehow into `TestingUtils`?

Reply:
We could add another resource/extension for this. We cant extend the executor resource because it may only contain a plain Executor.

Reply:
Although we couldn't put it into our test util modules.
Maybe we should just amend the FutureUtils to work against a plain ScheduledExecutorService.
Or add variants that also accept a ScheduledExecutorService.